pkg/scraper/client/resource/decode.go

```diff
@@ -44,7 +44,7 @@ func decodeBatch(b []byte, defaultTime time.Time, nodeName string) (*storage.Met
 	}
 	node := &storage.MetricsPoint{}
 	pods := make(map[apitypes.NamespacedName]storage.PodMetricsPoint)
-	parser, err := textparse.New(b, "", false, nil)
+	parser, err := textparse.New(b, "text/plain", "", false, false, nil)
```

Comment:
Due to changes in https://github.com/prometheus/prometheus/commit/421a3c22ea06d6966bff5127c65c94fd8855e190#diff-88b6f2039c68ed3e6c470170fb60854b4dc7ef381e14c690337e0db8d0a2aab9R138-R139 for v3

Reply:
@dgrisonnet  Do we want to specify a specific fallback scrape protocol?

Reply:
I don't think any would make sense. Metrics-server always scrapes kubelet's metrics in text format: https://github.com/kubernetes-sigs/metrics-server/blob/master/pkg/scraper/client/resource/client.go#L105-L137

No having any would also keep the same behavior as we had before this change in Prometheus: https://github.com/prometheus/prometheus/commit/421a3c22ea06d6966bff5127c65c94fd8855e190#diff-88b6f2039c68ed3e6c470170fb60854b4dc7ef381e14c690337e0db8d0a2aab9R138-R141

Reply:
ok, keeping only content type as current change then.